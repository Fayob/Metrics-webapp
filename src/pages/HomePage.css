.main_container {
  margin-top: 3rem;
  padding: 1rem;
  background: #f6516c;
}

.home_page_list {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 2px;
  margin: 1rem auto;
}

.home_page {
  cursor: pointer;
  background: url('../assets/covid_image.jpg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  height: 200px;
  min-width: 200px;
  padding: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-end;
}

.arrow_right {
  font-size: 22px;
}

.home_page_link {
  font-size: 15px;
  font-weight: 700;
}

.select {
  display: block;
  width: 50%;
  padding: 0.7rem;
  margin: 1rem auto 2rem;
}

.country_data {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

@media (min-width: 768px) {
  .home_page_list {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media (min-width: 1200px) {
  .home_page_list {
    grid-template-columns: repeat(4, 1fr);
  }
}
