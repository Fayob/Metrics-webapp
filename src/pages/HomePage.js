import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import Navbar from './Navbar';
import Heading from '../component/heading';
import Dropdown from '../component/Dropdown';
import Home from './Home';
import './HomePage.css';

const HomePage = () => {
  const [filter, setFilter] = useState('');

  const { worldCovid } = useSelector((state) => state.metrics);

  const onChangeHandler = (e) => {
    setFilter(e.target.value);
  };

  return (
    <>
      <Navbar navHeading="Total Cases" />
      <main className="main_container">
        <Heading name="global total cases" confirmedCases={(worldCovid.length > 0) ? worldCovid[0].TotalCases : 0} />
        <Dropdown onChangeHandler={onChangeHandler} />
        <Home filter={filter} />
      </main>
    </>
  );
};

export default HomePage;
