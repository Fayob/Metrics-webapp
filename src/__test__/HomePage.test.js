import React from 'react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
// We're using our own custom render function and not RTL's render.
import renderWithProviders from '../utils/utils';
import HomePage from '../pages/HomePage';
// We use msw to intercept the network request during the test,
// and return the response 'John Smith' after 150ms
// when receiving a get request to the `/api/user` endpoint
const handlers = [
  rest.get('api/metrics', (req, res, ctx) => res(ctx.json({
    isLoading: false,
    covid19Data: [{ Country: 'Nigeria', TotalCases: 123 }, { Country: 'UK', TotalCases: 456 }, { Country: 'Germany', TotalCases: 789 }, { Country: 'Canada', TotalCases: 245 }, { Country: 'USA', TotalCases: 123 }],
  }), ctx.delay(150))),
];

const server = setupServer(...handlers);

// Enable API mocking before tests.
beforeAll(() => server.listen());

// Reset any runtime request handlers we may add during the tests.
afterEach(() => server.resetHandlers());

// Disable API mocking after the tests are done.
afterAll(() => server.close());

describe('test redux and Homepage component', () => {
  test('render HomePage', () => {
    expect(renderWithProviders(<BrowserRouter><HomePage /></BrowserRouter>)).toMatchSnapshot();
  });
  test('fetches & receives a user after clicking the fetch user button', async () => {
    renderWithProviders(<BrowserRouter><HomePage /></BrowserRouter>);
    expect(screen.queryByText(/Africa/i)).toBeVisible();
    expect(screen.queryByText(/Asia/i)).toBeVisible();
    expect(screen.queryByText(/Europe/i)).toBeVisible();
  });
});
