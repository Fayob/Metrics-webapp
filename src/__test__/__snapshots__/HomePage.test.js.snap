// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test redux and Homepage component render HomePage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar_container"
      >
        <a
          class="arrow_back"
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
            />
          </svg>
        </a>
        <div>
          <p>
            Total Cases
          </p>
        </div>
        <div
          class="mic_setting"
        >
          <div
            class="mic"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
              />
              <path
                d="M19 10v2a7 7 0 0 1-14 0v-2"
              />
              <line
                x1="12"
                x2="12"
                y1="19"
                y2="23"
              />
              <line
                x1="8"
                x2="16"
                y1="23"
                y2="23"
              />
            </svg>
          </div>
          <div
            class="settings"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="3"
              />
              <path
                d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
              />
            </svg>
          </div>
        </div>
      </nav>
      <main
        class="main_container"
      >
        <div
          class="heading"
        >
          <div
            class="empty_div"
          />
          <div
            class="heading_div"
          >
            <h2
              class="heading_name"
            >
              global total cases
            </h2>
            <p>
              0
            </p>
          </div>
        </div>
        <select
          class="select"
        >
          <option
            class="option"
            value=""
          >
            All Continent
          </option>
          <option
            class="option"
            value="africa"
          >
             Africa 
          </option>
          <option
            class="option"
            value="asia"
          >
             Asia 
          </option>
          <option
            class="option"
            value="europe"
          >
             Europe 
          </option>
          <option
            class="option"
            value="northamerica"
          >
             North America 
          </option>
          <option
            class="option"
            value="southamerica"
          >
             South America 
          </option>
          <option
            class="option"
            value="australia"
          >
             Australian and Oceanian 
          </option>
        </select>
        <h2>
          Loading...
        </h2>
      </main>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar_container"
    >
      <a
        class="arrow_back"
        href="/"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </a>
      <div>
        <p>
          Total Cases
        </p>
      </div>
      <div
        class="mic_setting"
      >
        <div
          class="mic"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
            />
            <path
              d="M19 10v2a7 7 0 0 1-14 0v-2"
            />
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="23"
            />
            <line
              x1="8"
              x2="16"
              y1="23"
              y2="23"
            />
          </svg>
        </div>
        <div
          class="settings"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="3"
            />
            <path
              d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
            />
          </svg>
        </div>
      </div>
    </nav>
    <main
      class="main_container"
    >
      <div
        class="heading"
      >
        <div
          class="empty_div"
        />
        <div
          class="heading_div"
        >
          <h2
            class="heading_name"
          >
            global total cases
          </h2>
          <p>
            0
          </p>
        </div>
      </div>
      <select
        class="select"
      >
        <option
          class="option"
          value=""
        >
          All Continent
        </option>
        <option
          class="option"
          value="africa"
        >
           Africa 
        </option>
        <option
          class="option"
          value="asia"
        >
           Asia 
        </option>
        <option
          class="option"
          value="europe"
        >
           Europe 
        </option>
        <option
          class="option"
          value="northamerica"
        >
           North America 
        </option>
        <option
          class="option"
          value="southamerica"
        >
           South America 
        </option>
        <option
          class="option"
          value="australia"
        >
           Australian and Oceanian 
        </option>
      </select>
      <h2>
        Loading...
      </h2>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
